src/components/Main/Schedule/TableCell.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import Input from '@material-ui/core/Input';
+import { DataCellProps } from '../../../shared/types';
+
+export const DataCell = ({
+  value: initialValue,
+  row: { index },
+  column: { id },
+  updateMyData, // This is a custom function that we supplied to our table instance
+}: DataCellProps) => {
+  // We need to keep and update the state of the cell normally
+  const [value, setValue] = useState(initialValue);
+
+  const onChange = (e: any) => {
+    setValue(e.target.value);
+  };
+  const onBlur = () => {
+    updateMyData(index, id, value);
+  };
+
+  // Doesn't blur on enter
+  const handleEnter = (e: any) => {
+    e = e || window.event;
+    e.key === 'Enter' && updateMyData(index, id, value);
+  };
+
+  // If the initialValue is changed external, sync it up with our state
+  useEffect(() => {
+    setValue(initialValue);
+  }, [initialValue]);
+
+  return initialValue ? (
+    <Input
+      value={value}
+      onChange={onChange}
+      onBlur={onBlur}
+      onKeyUp={(e) => handleEnter(e)}
+    />
+  ) : (
+    ''
+  );
```

Comment:
My original thought was, why not just this: 
```js
  return <Input value={value} onChange={onChange} onBlur={onBlur} onKeyUp={(e) => handleEnter(e)} />;
```
But sounds like you don't always want it editable.

Reply:
Right. There'll be some cells which are empty. I'm on my phone now but tmr I can snap a pic to explain why. Or you could pull down the branch and run it and you'll see why. In the time column, there'll be empty cells where there'll be time data or empty cells to denote multiple duties during each time